src/RA_Achievement.cpp

```diff
@@ -54,149 +30,112 @@ void Achievement::Parse( const Value& element )
+
+	_ReadStringTil(sTemp, ':', pBuffer);
+	SetCreatedDate((time_t)atol(sTemp.c_str()));
+
+	_ReadStringTil(sTemp, ':', pBuffer);
+	SetModifiedDate((time_t)atol(sTemp.c_str()));
+
+	_ReadStringTil(sTemp, ':', pBuffer);
+	// SetUpvotes( (unsigned short)atol( sTemp.c_str() ) );
+
+	_ReadStringTil(sTemp, ':', pBuffer);
+	// SetDownvotes( (unsigned short)atol( sTemp.c_str() ) );
+
+	_ReadStringTil(sTemp, ':', pBuffer);
+	SetBadgeImage(sTemp);
 
-	//	buffer now contains TITLE : DESCRIPTION : $POINTS : $Author : $DateCreated : $DateModified : upvotes : downvotes : badgefilename
-	
-	pTitle				= _ReadStringTil( ':', pBuffer, TRUE );
-	pDesc				= _ReadStringTil( ':', pBuffer, TRUE );
-	pProgress			= _ReadStringTil( ':', pBuffer, TRUE );
-	pProgressMax		= _ReadStringTil( ':', pBuffer, TRUE );
-	pProgressFmt		= _ReadStringTil( ':', pBuffer, TRUE );
-	pAuthor				= _ReadStringTil( ':', pBuffer, TRUE );
-	
-	nPoints =			(unsigned int)atol(   _ReadStringTil( ':', pBuffer, TRUE ) );
-	nDateCreatedSecs =	(time_t)atol(		  _ReadStringTil( ':', pBuffer, TRUE ) );
-	nDateModifiedSecs =	(time_t)atol(		  _ReadStringTil( ':', pBuffer, TRUE ) );
-	nUpvotes =			(unsigned short)atol( _ReadStringTil( ':', pBuffer, TRUE ) );
-	nDownvotes =		(unsigned short)atol( _ReadStringTil( ':', pBuffer, TRUE ) );
-
-	pBadgeFilename		= _ReadStringTil( ':', pBuffer, TRUE );
-
-	SetPoints( nPoints );
-	SetCreatedDate( nDateCreatedSecs );
-	SetModifiedDate( nDateModifiedSecs );
-	SetTitle( pTitle );
-	SetDescription( pDesc );
-	SetAuthor( pAuthor );
-	SetProgressIndicator( pProgress );
-	SetProgressIndicatorMax( pProgressMax );
-	SetProgressIndicatorFormat( pProgressFmt );
-	//SetUpvotes( nUpvotes );
-	//SetDownvotes( nDownvotes );
-	SetBadgeImage( pBadgeFilename );
-	
 	return pBuffer;
 }
 
-BOOL Achievement::Test()
+static bool HasHitCounts(const ConditionSet& set)
 {
-	BOOL bDirtyConditions = FALSE;
-	BOOL bResetConditions = FALSE;
-
-	BOOL bRetVal = FALSE;
-	BOOL bRetValSubCond = NumConditionGroups() == 1 ? TRUE : FALSE;
-	for( size_t i = 0; i < NumConditionGroups(); ++i )
+	for (size_t i = 0; i < set.GroupCount(); i++)
 	{
-		if( i == 0 )
-			bRetVal = m_vConditions[i].Test( bDirtyConditions, bResetConditions, FALSE );
-		else	//	OR!
-			bRetValSubCond |= m_vConditions[i].Test( bDirtyConditions, bResetConditions, FALSE );
+		const ConditionGroup& group = set.GetGroup(i);
+		for (size_t j = 0; j < group.Count(); j++)
+		{
+			const Condition& condition = group.GetAt(j);
+			if (condition.CurrentHits() > 0)
+				return true;
+		}
 	}
 
+	return false;
+}
+
+BOOL Achievement::Test()
+{
+	bool bDirtyConditions = FALSE;
+	bool bResetConditions = FALSE;
+
+	bool bNotifyOnReset = GetPauseOnReset() && HasHitCounts( m_vConditions );
+
+	bool bRetVal = m_vConditions.Test( bDirtyConditions, bResetConditions );
+
 	if( bDirtyConditions )
 	{
 		SetDirtyFlag( Dirty_Conditions );
 	}
 
-	if( bResetConditions )
+	if( bResetConditions && bNotifyOnReset )
 	{
 		Reset();
```

Comment:
This `Reset` is unnecessary, `Test` will have already done the Reset. I'll correct in a future commit.

The `bNotifyOnReset` boolean that was added only pops the dialog if Pause on Reset it checked and at least one condition with a non-zero HitCount target as reset.